### PR TITLE
#212 - case studies page

### LINKED ISSUE #212: Case study page
Strona z listą wszystkich case studies
Link: https://xd.adobe.com/view/26f66876-cddd-469c-8b1f-b991dec40e55-73a5/screen/149a56ab-f403-4035-a608-4c0cbd75fcb7 

obraz

### LINKED ISSUE #213: Case study - details page
Strona ze szczegółami case studies
Link: https://xd.adobe.com/view/26f66876-cddd-469c-8b1f-b991dec40e55-73a5/screen/06a2e316-ea87-4408-91b0-dfa7260387ec/ 
Może być tak, że każde case study będzie wyglądało inaczej - więc trzeba to dobrze przemyśleć.
Do przemyślenia i przegadania z Kasią, jak mielibyśmy tym zarządzać.
(Ponieważ case studies nie będą się często zmieniać- to może jako gotowe strony do edycji w html?)
Temat jeszcze do szerszego omówienia

obraz

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
I created case studies page, single case study page and added changes from general review (that's the last time I'm mixing it to one pr, probably it would be better to add separated pr for those changes).

Case studies page:
image
Single case study base page - only presenting name, description and image of certain case study
image

It should close #212 and close #213 - for now we have this one details component, others will be created as needed.

Link to desktop design:
https://xd.adobe.com/view/26f66876-cddd-469c-8b1f-b991dec40e55-73a5/screen/eb45fea3-45c7-4e7e-a91e-d4c1ac710e4a/
and mobile design:
https://xd.adobe.com/view/48e3dd96-96c0-4801-aee3-93a97687467e-1db0/screen/9d2344b4-0163-4f41-941c-7b76aa2e286f/

### CODE REVIEW

File: app/Http/Controllers/CaseStudiesController.php [hunk 1]
```diff
@@ -4,18 +4,24 @@
...
 use Illuminate\Contracts\View\View;
 
 class CaseStudiesController extends Controller
 {
     public function index(Factory $factory): View
     {
-        return $factory->make("case_studies");
+        $caseStudies = CaseStudy::all()->where("published", true);
+
+        return $factory->make("case_studies", compact("caseStudies"));
```
Comment (kamilpiech97) [inline app/Http/Controllers/CaseStudiesController.php:17]:
```suggestion
        $caseStudies = CaseStudy::query()->where("published", true)->get();

        return $factory->make("case-studies/$view")
            ->with("caseStudies", $caseStudies);
```

Comment (kamilpiech97) [reply app/Http/Controllers/CaseStudiesController.php:17]: What do u think?

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Shadows on design are more visible:

Design:

image

Implementation:

image

2. Seeders took ages - I'm waiting something like 5 minutes and still seeders don't finish.

3. Mobile view - in design badges, titles and description are centered, but on implementation not:

Design:

image


Implementation:

image

4. On mobile view, the text "Pozwól nam przekształcić Twoją wizję w rzeczywistość..." is too big compared to design.

5. I see 403 in dev console

image

6. On design the button "Napisz do nas" is wider.

7. On case study details page, when I hover on button "Wróć do portfolio" it hovers on the next button.

image

File: tailwind.config.js [hunk 2]
```diff
@@ -16,8 +16,10 @@ module.exports = {
     extend: {
       colors: {
         gray: {
+          extraLight: '#f5f6fa',
```
[the remark this answers is not in the document]
Comment (AleksandraKozubal) [reply tailwind.config.js]: yes, changed all to uppercase

**Comment by AleksandraKozubal** [comment]
> 1. Shadows on design are more visible
> 
>  2. Seeders took ages - I'm waiting something like 5 minutes and still seeders don't finish.
> 
>  3. Mobile view - in design badges, titles and description are centered, but on implementation not:
> 
>  4. On mobile view, the text "Pozwól nam przekształcić Twoją wizję w rzeczywistość..." is too big compared to design.
> 
>  5. I see 403 in dev console
> 
>  6. On design the button "Napisz do nas" is wider.
> 
>   7. On case study details page, when I hover on button "Wróć do portfolio" it hovers on the next button.
>
1.fixed
2. still working on it
3.fixed
4. changed to smaller font on mobiles
5. Should be better
6. changed
7. fixed

File: app/Http/Controllers/CaseStudiesController.php (hunk 1 above)
Comment (AleksandraKozubal) [reply app/Http/Controllers/CaseStudiesController.php]: sure, changed

**Comment by AleksandraKozubal** [comment]
@EwelinaSkrzypacz I changed factories, now we're not using faker images, so seeder works faster.

**Review by EwelinaSkrzypacz — DISMISSED** [review]

File: lang/en/content.php [hunk 3]
```diff
@@ -57,4 +57,14 @@
     "about.section_4.subtitle_1" => "Our company promotes ",
     "about.section_4.subtitle_2" => "work-life balance, ",
     "about.section_4.subtitle_3" => "giving employees the opportunity to develop outside of the professional sphere as well. After hours, our team builds bonds through shared passions such as board games, mountain trips, basketball tournaments, and regular outdoor meetings.",
+    "case_studies.section_1.title_1" => "Our ",
+    "case_studies.section_1.title_2" => "projects",
+    "case_studies.section_1.subtitle_1" => "See what we worked on for our clients.",
+    "case_studies.section_1.subtitle_2" => "Each case study contains a detailed description of the project implementation.",
```
Comment (kamilpiech97) [inline lang/en/content.php:63]:
```suggestion
    "case_studies" => [
        "section_1" => [
            "title_2" => "projects",
            "subtitle_1" => "See what we worked on for our clients.",
            "subtitle_2" => "Each case study contains a detailed description of the project implementation."
        ],
    ],
```
maybe that should be more readable?

Comment (AleksandraKozubal) [reply lang/en/content.php]: formatted

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**